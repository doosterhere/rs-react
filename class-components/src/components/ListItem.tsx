import { FC } from 'react';

interface IListItemProps {
  name: string;
  // diameter: string;
  // terrain: string;
  // population: string;
}

const ListItem: FC<IListItemProps> = ({
  name,
  // diameter,
  // terrain,
  // population,
}) => {
  return (
    <div className="list-item">
      <div>
        <h3>{name}</h3>
      </div>
      {/* <div>
        <span>Diameter:</span> {diameter}
      </div>
      <div>
        <span>Terrain:</span> {terrain}
      </div>
      <div>
        <span>Population:</span> {population}
      </div> */}
    </div>
  );
};

export default ListItem;
