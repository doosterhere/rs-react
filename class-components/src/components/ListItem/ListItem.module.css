.item {
  width: 350px;
  padding: 15px 10px;
  border: 1px solid white;
  border-radius: 5px;

  font-size: 24px;
  color: black;

  box-shadow: 0 0 10px 0 rgba(0, 0, 0, 0.2);
  cursor: pointer;
  text-decoration: none;
}

.item:hover {
  box-shadow: 0 0 10px 0 rgba(0, 0, 0, 0.4);
}

.content {
  display: grid;
  justify-items: center;
  align-items: center;
  grid-template-columns: 1fr 8fr 1fr;
}

.input {
  margin: 0;
  width: 1.5rem;
  height: 1.5rem;
  cursor: pointer;
}

.input:hover {
  box-shadow: 0 0 10px 0 rgba(0, 0, 0, 0.4);
}
