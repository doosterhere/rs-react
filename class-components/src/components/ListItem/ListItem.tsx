import { FC } from 'react';
import { useParams, useSearchParams, NavLink } from 'react-router-dom';

import classes from './ListItem.module.css';

interface IListItemProps {
  name: string;
  id: string | undefined;
}

const ListItem: FC<IListItemProps> = ({ name, id }) => {
  const params = useParams();
  const [searchParams] = useSearchParams();

  return (
    <NavLink
      to={params.id === id ? `/?${searchParams.toString()}` : `/detail/${id}?${searchParams.toString()}`}
      className={classes.item}
    >
      {name}
    </NavLink>
  );
};

export { ListItem };
