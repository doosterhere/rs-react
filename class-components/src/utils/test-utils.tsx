// import { render } from '@testing-library/react';
// import { Provider } from 'react-redux';

// import { store } from '../store';
// import { ThemeProvider } from '../components';

// const renderWithRouter = (component: React.ReactNode, options: MemoryRouterProps = {}) => {
//   const Wrapper = ({ children }: { children: React.ReactNode }) => (
//     <MemoryRouter {...options}> {children} </MemoryRouter>
//   );

//   return render(component, { wrapper: Wrapper });
// };

// const renderWithProvider = (component: React.ReactNode, options: MemoryRouterProps = {}) => {
//   const Wrapper = ({ children }: { children: React.ReactNode }) => (
//     <Provider store={store}>
//       <ThemeProvider>
//         <MemoryRouter {...options}> {children} </MemoryRouter>
//       </ThemeProvider>
//     </Provider>
//   );

//   return render(component, { wrapper: Wrapper });
// };

// export { renderWithRouter, renderWithProvider };
