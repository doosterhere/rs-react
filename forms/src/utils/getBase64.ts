const getBase64 = (file: File): Promise<string> => {
  return new Promise(resolve => {
    const reader = new FileReader();
    reader.onloadend = () => {
      const base64String = reader.result?.toString() || '';
      resolve(base64String);
    };

    reader.readAsDataURL(file);
  });
};

export { getBase64 };
