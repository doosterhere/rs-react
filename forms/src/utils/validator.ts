import { MIN_AGE, MAX_AGE, MAX_FILE_SIZE, MB } from '@/constants';

type errors = { [key: string]: string };

const validator = {
  name: (value: string | null, errors: errors): void => {
    if (!value) {
      errors.name = 'Name is required';
      return;
    }

    if (value.length < 3) {
      errors.name = 'Name must be at least 3 characters long';
      return;
    }

    if (!/^[A-Za-z\s-]+$/.test(value)) {
      errors.name = 'Name should not contain special characters';
      return;
    }

    if (!/^[A-Z]{1}[A-Za-z\s-]+$/.test(value)) {
      errors.name = 'The name must begin with a capital letter';
      return;
    }

    delete errors.name;
  },
  age: (value: number | null, errors: errors): void => {
    if (!value && value !== 0) {
      errors.age = 'Age is required';
      return;
    }

    if (value < 1 || value > 130) {
      errors.age = `Age must be between ${MIN_AGE} and ${MAX_AGE}`;
      return;
    }

    delete errors.age;
  },
  email: (value: string | null, errors: errors): void => {
    if (!value) {
      errors.email = 'Email is required';
      return;
    }

    if (!/^[^$!#^\-_*'%?]*[a-z0-9\-_.]{1,64}@[a-z0-9.-]{1,253}\.[a-z]{2,}$/i.test(value)) {
      errors.email = 'Please enter a valid email';
      return;
    }

    delete errors.email;
  },
  password: (value: string | null, errors: errors): void => {
    if (!value) {
      errors.password = 'Password is required';
      return;
    }

    if (value.length < 8) {
      errors.password = 'Password must be at least 8 characters long';
      return;
    }

    if (!/[A-Z]/.test(value)) {
      errors.password = 'At least one uppercase letter is required';
      return;
    }

    if (!/[a-z]/.test(value)) {
      errors.password = 'At least one lowercase letter is required';
      return;
    }

    if (!/[0-9]/.test(value)) {
      errors.password = 'At least one number is required';
      return;
    }

    if (!/[!@#$%^&*_.]/.test(value)) {
      errors.password = 'At least one special character is required';
      return;
    }

    delete errors.password;
  },
  passwordRepeat: (value: string | null, comparedValue: string | null, errors: errors): void => {
    if (!value) {
      errors.passwordRepeat = 'Password confirmation is required';
      return;
    }

    if (value !== comparedValue) {
      errors.passwordRepeat = "Passwords don't match";
      return;
    }

    delete errors.passwordRepeat;
  },
  gender: (value: string | null, errors: errors): void => {
    if (!value) {
      errors.gender = 'Please choose your gender';
      return;
    }

    delete errors.gender;
  },
  country: (value: string | null, errors: errors): void => {
    if (!value) {
      errors.country = 'Please choose your country';
      return;
    }

    delete errors.country;
  },
  picture: (value: FileList | null, errors: errors): void => {
    if (value && value.length === 0) {
      errors.picture = 'Please upload a file';
      return;
    }

    if (value && value[0] && value[0].size > MAX_FILE_SIZE) {
      errors.picture = `Max allowed size is ${MB}Mb`;
      return;
    }

    if (value && value[0] && !['image/png', 'image/jpeg', '.jpeg', '.png'].includes(value[0].type)) {
      errors.picture = 'Only PNG and JPEG files are accepted';
      return;
    }

    delete errors.picture;
  },
  agreed: (value: boolean | null, errors: errors): void => {
    if (!value) {
      errors.agreed = 'Please accept Terms and Conditions';
      return;
    }

    delete errors.agreed;
  },
};

export { validator };
