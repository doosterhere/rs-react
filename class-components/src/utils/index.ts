import { renderWithRouter } from './test-utils';

export { renderWithRouter };
