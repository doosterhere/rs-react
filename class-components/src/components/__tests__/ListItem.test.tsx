import { screen } from '@testing-library/react';

import { renderWithRouter } from '../../utils';

import ListItem from '../ListItem';

describe('ListItem', () => {
  it('should renders correctly', () => {
    const props = {
      name: 'test',
      id: '1',
    };

    renderWithRouter(<ListItem {...props} />);

    expect(screen.getByText(/test/i)).toBeInTheDocument();

    expect(screen.getByRole('link')).toMatchSnapshot();
  });
});
