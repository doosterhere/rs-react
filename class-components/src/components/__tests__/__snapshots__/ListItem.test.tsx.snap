// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ListItem should renders correctly 1`] = `
<a
  class="list-item"
  href="/detail/1"
>
  test
</a>
`;
